ioctl/newcmd/account/accountupdate.go

```diff
@@ -0,0 +1,141 @@
+		config.Chinese: "为IoTeX账户更新密码",
+	}
+	updateCmdUses = map[config.Language]string{
+		config.English: "update [ALIAS|ADDRESS]",
+		config.Chinese: "update [别名|地址]",
+	}
+)
+
+// NewAccountUpdate represents the account update command
+func NewAccountUpdate(client ioctl.Client) *cobra.Command {
+	use, _ := client.SelectTranslation(updateCmdShorts)
+	short, _ := client.SelectTranslation(updateCmdUses)
+
+	cmd := &cobra.Command{
+		Use:   use,
+		Short: short,
+		Args:  cobra.RangeArgs(0, 1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SilenceUsage = true
+			arg := ""
+			if len(args) == 1 {
+				arg = args[0]
+			}
+
+			acc, err := client.GetAddress(arg)
+			if err != nil {
+				return output.NewError(output.AddressError, "", err)
+			}
+			addr, err := address.FromString(acc)
+			if err != nil {
+				return output.NewError(output.ConvertError, "failed to convert string into addr", err)
+			}
+
+			if CryptoSm2 {
+				// find the pem file and update
+				filePath, err := findSm2PemFile(addr)
+				if err != nil {
+					return output.NewError(output.ReadFileError, fmt.Sprintf("crypto file of account #%s not found", addr), err)
+				}
+				if err = readSecretAndUpdate(client, acc,
+					func(currentPassword string) error {
+						_, err = crypto.ReadPrivateKeyFromPem(filePath, currentPassword)
+						if err != nil {
+							return output.NewError(output.CryptoError, "error occurs when checking current password", err)
+						}
+						return nil
+					},
+					func(currentPassword, newPassword string) error {
+						if err := crypto.UpdatePrivateKeyPasswordToPem(filePath, currentPassword, newPassword); err != nil {
+							return output.NewError(output.KeystoreError, "failed to update pem file", err)
+						}
+						return nil
+					},
+				); err != nil {
+					return err
+				}
+			} else {
+				// find the keystore and update
+				ks := client.NewKeyStore(config.ReadConfig.Wallet, keystore.StandardScryptN, keystore.StandardScryptP)
+				for _, v := range ks.Accounts() {
+					if bytes.Equal(addr.Bytes(), v.Address.Bytes()) {
+						if err = readSecretAndUpdate(client, acc,
+							func(currentPassword string) error {
+								_, err = ks.SignHashWithPassphrase(v, currentPassword, hash.ZeroHash256[:])
+								if err != nil {
+									return output.NewError(output.KeystoreError, "error occurs when checking current password", err)
+								}
+								return nil
+							},
+							func(currentPassword, newPassword string) error {
+								if err := ks.Update(v, currentPassword, newPassword); err != nil {
+									return output.NewError(output.KeystoreError, "failed to update keystore", err)
+								}
+								return nil
+							},
+						); err != nil {
+							return err
+						}
+					}
+				}
+			}
+			output.PrintResult(fmt.Sprintf("Account #%s has been updated.", acc))
+			return nil
+		},
+	}
+	return cmd
+}
+
+func readSecretAndUpdate(client ioctl.Client, acc string,
```

Comment:
@Liuhaai I put the common logic into this. The caller will pass down the password checker and password updater into this sub-function.